controllers/alphacontrollers/outboundwebhook_controller.go

```diff
@@ -0,0 +1,453 @@
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.4/pkg/reconcile
+var (
+	outboundWebhookFinalizerName = "outbound-webhook.coralogix.com/finalizer"
+)
+
+func (r *OutboundWebhookReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	var (
+		resultError = ctrl.Result{RequeueAfter: 40}
+		resultOk    = ctrl.Result{RequeueAfter: defaultErrRequeuePeriod}
+		err         error
+	)
+
+	log := log.FromContext(ctx).WithValues(
+		"outboundWebhook", req.NamespacedName.Name,
+		"namespace", req.NamespacedName.Namespace,
+	)
+
+	outboundWebhook := &coralogixv1alpha1.OutboundWebhook{}
+	if err = r.Client.Get(ctx, req.NamespacedName, outboundWebhook); err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return ctrl.Result{}, nil
+		}
+		return resultError, err
+	}
+
+	if ptr.Deref(outboundWebhook.Status.ID, "") == "" {
+		err = r.create(ctx, log, outboundWebhook)
+		if err != nil {
+			log.Error(err, "Error on creating outbound-webhook")
+			return resultError, err
+		}
+		return resultOk, nil
+	}
+
+	if !outboundWebhook.ObjectMeta.DeletionTimestamp.IsZero() {
+		err = r.delete(ctx, log, outboundWebhook)
+		if err != nil {
+			log.Error(err, "Error on deleting outbound-webhook")
+			return resultError, err
+		}
+		return resultOk, nil
+	}
+
+	err = r.update(ctx, log, outboundWebhook)
+	if err != nil {
+		log.Error(err, "Error on updating outbound-webhook")
+		return resultError, err
+	}
+
+	return resultOk, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *OutboundWebhookReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&coralogixv1alpha1.OutboundWebhook{}).
+		Complete(r)
+}
+
+func (r *OutboundWebhookReconciler) create(ctx context.Context, log logr.Logger, webhook *coralogixv1alpha1.OutboundWebhook) error {
+	createRequest, err := webhook.ExtractCreateOutboundWebhookRequest()
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Error to extract create-request out of the outbound-webhook -\n%v", webhook))
+		return err
+	}
+
+	log.V(1).Info(fmt.Sprintf("Creating outbound-webhook-\n%s", protojson.Format(createRequest)))
+	createResponse, err := r.OutboundWebhooksClient.CreateOutboundWebhook(ctx, createRequest)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Received an error while creating outbound-webhook -\n%s", protojson.Format(createRequest)))
+		return err
+	}
+	log.Info(fmt.Sprintf("outbound-webhook was created-\n%s", protojson.Format(createResponse)))
+
+	webhook.Status = coralogixv1alpha1.OutboundWebhookStatus{
+		ID:                  ptr.To(createResponse.Id.GetValue()),
```

Comment:
I don't think we should store this fields on status, only id.
This fields are already present on the spec and we can get if from there if needed

Reply:
I don't have solid opinion about it yet, but in this case some of the fields are "computed" (will be determined by the backend), so it can be helpful to keep also the status.